core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -170,6 +170,8 @@ static CloseableIterable<FileScanTask> planFiles(StaticTableScan scan) {
                       scan.filter(), transformedSpec, caseSensitive);
                 });
 
+    // Note, the delete files aren't applied to the data files so if there are partition values that
+    // have been updated then both the 'old' and the 'new' values are present in the output.
```

Comment:
Im actually not sure it should be inline comment, those to me seems for impl details for specific lines and maybe not read if you are just skimming.  If we need to, how about the javadoc of the class or method, which is more about the overall behavior, as is being commented here?

And also if we do keep it, we should change to document the general case, as mentioned in the comment on the docs page, like:

`Note, the delete files aren't applied to the data files, so partitions with 0 rows may still be present in the output.`

Reply:
Understood. I removed the comments from this file for now. However, I feel that we should have a way somehow to document the behaviour of the metadata table APIs that is not Spark specific. I'm open for suggestions, though, as I don't want to duplicate the remarks on the [Spark/Queries](https://iceberg.apache.org/docs/latest/spark-queries/#partitions) page either, but in fact they might be useful in the PartitionsTable.java file as well.

Reply:
Yea I was trying to say, maybe on the class itself?  (Initially thinking method, but it's a overriden one)  Anyway change looks good to me as is, up to you if you want to put it on there.